src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -307,3 +307,25 @@ class OverlapsClauseSegment(BaseSegment):
 
     type = "overlaps_clause"
     match_grammar = Nothing()
+
+
+@tsql_dialect.segment(replace=True)
+class LimitClauseSegment(BaseSegment):
+    """A `LIMIT` clause like in `SELECT`.
+
+    Not present in T-SQL.
+    """
+
+    type = "limit_clause"
+    match_grammar = Nothing()
+
+
+@tsql_dialect.segment(replace=True)
+class NamedWindowSegment(BaseSegment):
+    """A WINDOW clause.
+
+    Not present in T-SQL.
+    """
+
+    type = "named_window"
+    match_grammar = Nothing()
```

Comment:
I'm not sure exactly what you are trying to achieve here but not sure litering the dialect with all these `Nothing()`s is the way to go about it.

Surely you would be better instead to update the segments referencing them?:

```suggestion


@tsql_dialect.segment(replace=True)
class UnorderedSelectStatementSegment(BaseSegment):
    """A `SELECT` statement without any ORDER clauses or later.

    We need to change ANSI slightly
    """

    type = "select_statement"
    # Remove the LimitClauseSegment and NamedWindowSegment from ANSI
    match_grammar = StartsWith(
        Ref("SelectClauseSegment"),
        terminator=OneOf(
            Ref("SetOperatorSegment"),
            Ref("WithNoSchemaBindingClauseSegment"),
            Ref("OrderByClauseSegment"),
        ),
        enforce_whitespace_preceeding_terminator=True,
    )

    parse_grammar = ansi_dialect.get_segment(
        "UnorderedSelectStatementSegment"
    ).match_grammar.copy()


@tsql_dialect.segment(replace=True)
class SelectStatementSegment(BaseSegment):
    """A `SELECT` statement.

    We need to change ANSI slightly
    """

    type = "select_statement"
    match_grammar = ansi_dialect.get_segment(
        "SelectStatementSegment"
    ).match_grammar.copy()

    # Remove the Limit and Window statements from ANSI
    parse_grammar = UnorderedSelectStatementSegment.parse_grammar.copy(
        insert=[
            Ref("OrderByClauseSegment", optional=True),
        ]
    )
```
 
Note above code also requires importing `StartsWith`

Reply:
That approach is an option.  My consideration is that overwriting more central segments like UnorderedSelectStatementSegment is a bigger code change.  If two or three months from now UnorderedSelectStatementSegment receives a rewrite for some reason -- performance, breakfix, functionality -- the TSQL dialect wouldn't receive the benefits of that rewrite.  On the other hand, of course, TSQL wouldn't receive any negative impacts of that rewrite.  Adding the Nothing() stubs feels more elegant to me, but I'm the neophyte here regarding both python and SQLFluff.  Let me know if any of that makes sense.

Reply:
I think for complex segments you're right. But in that case, it could also be refactored to make it easier to overwrite only certain functions in different dialects.

For simple change like here - where the change is almost the same size as the other proposal, I think this is better. And avoids the need for the change to the `Nothing()` segment above.

The `Nothing()` approach feels like a patch on top of a patch, so should be used sparingly IMHO (we have very few instances of these, and mostly for simple Grammars).